src/thumbnailer.h

```diff
@@ -64,17 +65,29 @@ class Thumbnailer {
 
   // Returns the PSNR value of a re-encoded WebPPicture at given config.
   // Returns -1 on errors.
-  int GetPSNR(WebPPicture* const pic, const WebPConfig& config);
+  float GetPSNR(WebPPicture* const pic, const WebPConfig& config);
+
+  // Returns the size of a re-encoded WebPPicture at given config in bytes.
+  // Returns -1 on errors.
+  int GetSize(WebPPicture* const pic, const WebPConfig& config);
 
   // Generates the animation so that all frames have similar PSNR (all) values.
   // In case of failure, takes the animation generated by GenerateAnimation() as
   // result.
   Status GenerateAnimationEqualPSNR(WebPData* const webp_data);
 
+  // Tries near-lossless to encode each frame. In case of failure, keeps the
+  // latest lossy encoded frame.
```

Comment:
You should add that either GenerateAnimation or GenerateAnimationEqualPSNR must be called before this (it's important to document how the API must be used).